src/pages/academy/game/Game.tsx

```diff
@@ -50,9 +52,93 @@ function Game() {
     }
   }, [session, achievements, goals]);
 
+  // This is a custom hook imported from @mantine/hooks that handles the fullscreen logic
+  // It returns a ref to attach to the element that should be fullscreened,
+  // a function to toggle fullscreen and a boolean indicating whether the element is fullscreen
+  const {
+    ref: fullscreenRef,
+    toggle: toggleFullscreen,
+    fullscreen: isFullscreen
+  } = useFullscreen<HTMLDivElement>();
+
+  // This function is a wrapper around toggleFullscreen that also locks the screen orientation
+  // to landscape when entering fullscreen and unlocks it when exiting fullscreen
+  const enhancedToggleFullscreen = async () => {
+    toggleFullscreen();
+
+    if (window.screen.orientation) {
+      if (!isFullscreen) {
+        window.screen.orientation.lock('landscape');
+      } else {
+        window.screen.orientation.unlock();
+      }
+    }
+  };
+
+  const gameDisplayRef = React.useRef<HTMLDivElement | null>(null);
```

Comment:
What is the purpose of this `gameDisplayRef` here, can we not just use the ref provided by the `useFullscreen` hook?

Reply:
I don't think so. The `gameDisplayRef` is a React RefObject that I used in line 99 to 113 to handle the resizing of the game display when entering/exiting fullscreen. But the ref provided by the `useFullscreen` hook is a callback ref function. See [https://mantine.dev/hooks/use-fullscreen/#definition](url).

Reply:
My question is why do you need to handle the resizing manually? Doesn't `useFullscreen` already handle it for you?

Reply:
The fullscreen Icon element is a child of the game-display div element, which the Phaser game engine inserts the game canvas element in. The game canvas element has a fixed aspect ratio and its size is determined by the size of the game-display div element. When browser window width is too wide and height is too small, there will be empty space on the left and right of the game canvas element. So the left offset of the fullscreen Icon element is updated to ensure that it remains aligned with the top left of the game canvas element.

Reply:
I think the solution is very hacky. Why not just rely on the DOM to position the icon button correctly instead of using a resize observer? Instead of putting the button as a child you can just put it as a sibling, the wrap a parent container around both.

Maybe something like:

```tsx
{/* parent, relative, margin auto to squeeze child + align center */}
<div style={{ position: 'relative' }}>
  {/* absolute, top = left = padding */}
  <Icon /* ... */ />
  <div id="game-display"></div>
</div>
```

Reply:
Then you don't need to use any resize logic cause the DOM does the layouting for you

Reply:
I tried using `margin: auto` with the Icon button both as a child and as a sibling with a parent container wrapper. However, this causes issues with the resizing of the game canvas element since it references the game-display div element's width and height to determine its width and height. So, when you increase the size of the browser window, more margin gets added instead of the game canvas element getting enlarged.

Reply:
Ok, I understand it now; I still feel the solution is hacky. After looking at game code, I have an idea for a refactor. Let me discuss with @lhw-1 first.